example/src/PreJoinPage.tsx

```diff
@@ -189,11 +191,14 @@ export const PreJoinPage = () => {
               onClick={toggleAudio}
               onSourceSelected={setAudioDevice}
             />
-            <VideoSelectButton
-              isEnabled={videoTrack !== undefined}
-              onClick={toggleVideo}
-              onSourceSelected={selectVideoDevice}
-            />
+            {/** check for acquired video first, as iOS Safari seems to drop the previously acquired video track after trying to acquire again here */}
+            {acquiredVideo && (
+              <VideoSelectButton
```

Comment:
I think the reason it's failing is because they are being called simultaneously... the path for not acquiring them if they already were, is already in place.

Reply:
hmm, I see the code trying to prevent duplicate acquisitions in there.. but still a bit unclear how this change addresses simultaneous acquisitions. For example:

`createLocalVideoTrack` will only set `acquiredVideo` to false after its promise resolves. so if both are trying to acquire getUserMedia at the same time, it would still fail even with this change.

Reply:
the idea here war to delay the second call to `getDevices` until `acquiredVideo` is true, by only rendering the `VideoSelectComponent` (which automatically calls `getDevices`) afterwards.
You are right, that it the solution is rather application specific. Will try to find a way around it.

Reply:
found the related safari bug: https://bugs.webkit.org/show_bug.cgi?id=179363

The only other option I see to do it on the SDK level, is to track calls to getUserMedia within the SDK, which would lead to that tracking var hanging around globally as `createLocalTracks` are both independent from any `Room` instance.